### PR TITLE
initial GraphQL component examples

### PR DESCRIPTION
**Pull request by mburr-salesforce**
### What does this PR do?
Adds a new GraphQL tab to the recipes app with 3 example components:

- graphqlContacts fetches a set of Contact records using the GraphQL `@wire` and displays them
- graphqlVariables implements a basic search of Contact records to show how to use GraphQL variables with `@wire`
- graphqlPagination is a simple paginated display of Contacts build on the GraphQL `@wire`

### What issues does this PR fix or reference?
It shows how to use the `graphql` wire adapter in beta starting in the Summer '23 release.

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before
No examples of how to use GraphQL with LWC components.

### Functionality After
3 example components that use GraphQL to fetch record data from the server.

### CODE REVIEW

**Comment by mburr-salesforce** [comment]
@pozil I wanted to get some feedback on the components I've added here before I spend time working on tests. Happy to do a huddle with you to go over some of the finer points & show you what the UI looks like.

**Comment by mburr-salesforce** [comment]
@pozil cleaned up a few odds & ends and added tests. Need to chat with you re: when would be the right time to merge this.

File: force-app/main/default/lwc/graphqlPagination/__tests__/graphqlPagination.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { createElement } from 'lwc';
+import GraphqlPagination from 'c/graphqlPagination';
+import { graphql } from 'lightning/uiGraphQLApi';
+
+// Mock realistic data
+const mockGraphQL = require('./data/graphqlPaginationResponse.json');
+
+describe('c-graphql-pagination', () => {
```
Comment (pozil) [inline force-app/main/default/lwc/graphqlPagination/__tests__/graphqlPagination.test.js:8]: @mburr-salesforce shouldn't we add a test that covers navigation to the next page and check the following:
1. we use `endCursor` correctly in the next query
2. the page counter increments

File: force-app/main/default/flexipages/GraphQL.flexipage-meta.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <itemInstances>
+            <componentInstance>
+                <identifier>graphqlPagination</identifier>
+<componentName>graphqlPagination</componentName>
+            </componentInstance>
+        </itemInstances>
+        <name>region4</name>
+        <type>Region</type>
+    </flexiPageRegions>
+    <masterLabel>GraphQL</masterLabel>
```
Comment (pozil) [inline force-app/main/default/flexipages/GraphQL.flexipage-meta.xml:37]: Let's indicate that this feature is Beta for now:
```suggestion
    <masterLabel>GraphQL (Beta)</masterLabel>
```

**Review by pozil — CHANGES_REQUESTED** [review]
Looking great, thanks @mburr-salesforce.
I left some minor comments. Can you address them and run prettier?

**Review by pozil — CHANGES_REQUESTED** [review]
Looking great, thanks @mburr-salesforce.
I left some minor comments. Can you address them and run prettier?

**Review by pozil — CHANGES_REQUESTED** [review]
Looking great, thanks @mburr-salesforce.
I left some minor comments. Can you address them and run prettier?

**Review by pozil — CHANGES_REQUESTED** [review]
Looking great, thanks @mburr-salesforce.
I left some minor comments. Can you address them and run prettier?

File: force-app/main/default/flexipages/GraphQL.flexipage-meta.xml (hunk 2 above)
Comment (mburr-salesforce) [reply force-app/main/default/flexipages/GraphQL.flexipage-meta.xml]: I updated this, but it appears to actually be the tab label that gets displayed.

File: force-app/main/default/lwc/graphqlPagination/__tests__/graphqlPagination.test.js (hunk 1 above)
Comment (mburr-salesforce) [reply force-app/main/default/lwc/graphqlPagination/__tests__/graphqlPagination.test.js:9]: I added a new test that advances to the next page & added checks the states of both buttons & the status text. LMK if there are additional things you'd like to have the test check.

**Comment by mburr-salesforce** [comment]
> I left some minor comments. Can you address them and run prettier?

@pozil done, although the only thing prettier changed was some html wrapping.

**Comment by pozil** [comment]
Thanks a lot for the updates, we're almost there.

> @pozil done, although the only thing prettier changed was some html wrapping.

Yes, I agree that it's not a critical formatting change in this case but it puts CI in the green so it's needed.
Talking about CI: we now need to wait for a release of `sfdx-lwc-jest` that includes your fix.

**Comment by mburr-salesforce** [comment]
@pozil A new prerelease `sfdx-lwc-jest` was published to npm today. I updated the dependencies to pick it up, but `npm install` upgraded a ton of other stuff as well. Not sure how you want to proceed?

**Pull request merged**